libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -1362,106 +1365,571 @@ def _extract_images_from_page(self, page: pypdfium2._helpers.page.PdfPage) -> st
+                        # and yield the table
+                        used_arrays[i] = True
+                        # print(f"yield table {i}")
+                        yield tables_content[i]
+                    break
+            if not is_table:
+                # print(f'  Add {word["text"]}')
+                extract_wordmaps.append((word, o))
+        if extract_wordmaps:
+            # Text after the array ?
+            new_wordmap = text.WordMap(tuples=extract_wordmaps)
+            new_textmap = new_wordmap.to_textmap(
+                **{k: kwargs[k] for k in text.TEXTMAP_KWARGS if k in kwargs}
+            )
+            # print(f"yield {new_textmap.to_string()}")
+            yield new_textmap.to_string()
+        # Add images-
+        for content in images_content:
+            yield content
+
+    def _extract_images_from_page(self, page: pdfplumber.page.Page) -> list[np.ndarray]:
+        """Extract images from a PDF page.
+
+        Args:
+            page: The PDF page to extract images from.
+
+        Returns:
+            A list of extracted images as numpy arrays.
+        """
         from PIL import Image
 
-        if not self.extract_images:
-            return ""
+        if not self.images_parser:
+            return []
 
         images = []
         for img in page.images:
-            if img["stream"]["Filter"].name in _PDF_FILTER_WITHOUT_LOSS:
-                if img["stream"]["BitsPerComponent"] == 1:
-                    images.append(
-                        np.array(
-                            Image.frombytes(
-                                "1",
-                                (img["stream"]["Width"], img["stream"]["Height"]),
-                                img["stream"].get_data(),
-                            ).convert("L")
-                        )
-                    )
-                else:
+            if "Filter" in img["stream"]:
+                if img["stream"]["Filter"].name in _PDF_FILTER_WITHOUT_LOSS:
                     images.append(
                         np.frombuffer(img["stream"].get_data(), dtype=np.uint8).reshape(
                             img["stream"]["Height"], img["stream"]["Width"], -1
                         )
                     )
-            elif img["stream"]["Filter"].name in _PDF_FILTER_WITH_LOSS:
-                images.append(img["stream"].get_data())
-            else:
-                warnings.warn("Unknown PDF Filter!")
+                elif img["stream"]["Filter"].name in _PDF_FILTER_WITH_LOSS:
+                    buf = np.frombuffer(img["stream"].get_data(), dtype=np.uint8)
+                    images.append(
+                        np.array(Image.open(io.BytesIO(buf.tobytes())))  # type: ignore
+                    )
+                else:
+                    logger.warning("Unknown PDF Filter!")
+
+        return images
+
+    def _extract_tables_bbox_from_page(
+        self,
+        page: pdfplumber.page.Page,
+    ) -> list[tuple]:
```

Comment:
tuple[float,float,float,float] ? 

Could you document the mapping between the type and the coordinate system? different coordinate systems are used by different parsers

i.e., 

top, left, bottom, right
left, top, right, bottom
left, top, width, height

and position can be in pixels or sometimes in percentage w/ respect to the container

Reply:
it's private for PDFPlumber